src/ophyd_async/epics/motion/motor.py

```diff
@@ -95,3 +187,47 @@ async def stop(self, success=False):
         # Put with completion will never complete as we are waiting for completion on
         # the move above, so need to pass wait=False
         await self.motor_stop.trigger(wait=False)
+
+    async def _prepare_velocity(
+        self, start_position: float, end_position: float, time_for_move: float
+    ) -> float:
+        fly_velocity = (start_position - end_position) / time_for_move
+        max_speed = await self.max_velocity.get_value()
+        egu = await self.motor_egu.get_value()
+        if abs(fly_velocity) > max_speed:
```

Comment:
Nit: The term speed is not the same as the term velocity... But do we care?

Reply:
Yeah, the motor signal should really be called `max_speed`, but I didn't think this was worth changing this